pkg/controller/command/command_controller.go

```diff
@@ -189,20 +209,33 @@ func (r *ReconcileCommand) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, nil
 	}
 
-	commandConfigName := command.Name + "-command-configmap"
-	ips := command.Status.IPs
-	if len(ips) == 0 {
-		ips = []string{"0.0.0.0"}
+	config, err := r.getConfig(command)
+	if err != nil {
+		return reconcile.Result{}, err
 	}
-	if err = r.configMap(commandConfigName, "command", command, adminPasswordSecret, swiftSecret).ensureCommandConfigExist(ips[0], keystoneAddress, keystonePort, keystoneAuthProtocol, psql.Status.Endpoint); err != nil {
+	if err = r.kubernetes.Owner(command).EnsureOwns(config); err != nil {
 		return reconcile.Result{}, err
 	}
+	if config.Status.Endpoint == "" {
+		return reconcile.Result{}, nil
+	}
 
+	podIPs := []string{"0.0.0.0"}
 	if len(commandPods.Items) > 0 {
 		err = contrail.SetPodsToReady(commandPods, r.client)
 		if err != nil {
 			return reconcile.Result{}, err
 		}
+		podIPs = nil
+	}
+
+	commandConfigName := command.Name + "-command-configmap"
+
+	for _, pod := range commandPods.Items {
+		podIPs = append(podIPs, pod.Status.PodIP)
+	}
+	if err = r.configMap(commandConfigName, "command", command, adminPasswordSecret, swiftSecret).ensureCommandConfigExist(keystonePort, podIPs[0], keystoneAddress, keystoneAuthProtocol, psql.Status.Endpoint, config.Status.Endpoint); err != nil {
```

Comment:
why we are generating only one config file with one podIP as hostIP? shouldn't it be generated per POD?

Reply:
We should change generation of files to be done for all of the pods moreover there will be changes in endpoint registration. As we agreed we can do this in a seperate PR since Command is not HA-ready right now

Reply:
Yes.I will create separate PR for the mentioned changes.